vm/src/vm.rs

```diff
@@ -1138,6 +1158,21 @@ impl VirtualMachine {
     pub fn current_exception(&self) -> Option<PyObjectRef> {
         self.exceptions.borrow().last().cloned()
     }
+
+    pub fn check_signals(&self) {
+        for (signum, handler) in self.signal_handlers.borrow().iter() {
+            if *signum as usize >= NSIG {
+                panic!("Signum bigger then NSIG");
+            }
+            let triggerd;
+            unsafe {
```

Comment:
Please move this unsafe code into the signal module. This function should just loop over the list of triggered events. Also, you probably want to guard this for loop with some sort of lock. Another solution would be to use the `std::sync::mpsc` fifo, which is thread safe. This might be handy, so that the signal handlers still can add events, while this loop still is processing.

Reply:
There is not problem that the an event will be added while I process the list. I will just process it in the next run.